### PR TITLE
Fix usage or threadInfoStruct JS variable that was removed in #12853 

### PR DESCRIPTION
**Pull request by sbc100**
This was supposed to be removed in #12853 but it was still
be used in the 'cencel' and 'processThreadQueue' message
handlers.  Because of the way JS `var` is hoisted to function
scope a stale version of this variable was bring read.

### CODE REVIEW

**Comment by sbc100** [comment]
Still looking for a way to test this.  

cc @walkingeyerobot

**Comment by sbc100** [comment]
Adapted test from https://gist.github.com/walkingeyerobot/7b44b9e24c2ceac9bff7632d143176a7.  PTAL

**Review by kripken — APPROVED** [review]
I'm not entirely sure why the test checks this specific change, but it fails without the fix? Fix lgtm, so test lgtm too if so.

**Pull request merged**
